pkg/controllers/nodetemplate/controller.go

```diff
@@ -110,3 +122,26 @@ func (c *Controller) resolveSecurityGroup(ctx context.Context, nodeTemplate *v1a
 
 	return nil
 }
+
+func (c *Controller) resolveAMI(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate) error {
+	amiFamily := amifamily.GetAMIFamily(nodeTemplate.Spec.AMIFamily, &amifamily.Options{})
+	instancetypes, err := c.instanceTypesProvider.List(ctx, &v1alpha5.KubeletConfiguration{}, nodeTemplate)
+	if err != nil {
+		return err
+	}
+
+	amis, err := c.amiProvider.GetAMIWithRequirements(ctx, nodeTemplate, instancetypes, amiFamily)
```

Comment:
This function shouldn't depend on instance types. We ideally want to decouple this requirement resolution from instance types entirely